website/docs/docs/dbt-cloud-apis/sl-python-sdk.md

```diff
@@ -0,0 +1,138 @@
+---
+title: "Python SDK"
+id: sl-python
+description: "Learn how to use the dbt Semantic Layer SDK for Python library to interact with the dbt Semantic Layer."
+tags: [Semantic Layer, APIs]
+keywords: [dbt Cloud, API, dbt Semantic Layer, python]
+sidebar_label: "Python SDK"
+---
+
+The [`dbt-sl-sk` Python software development kit](https://github.com/dbt-labs/semantic-layer-sdk-python) (SDK) is a Python library that provides you with easy access to the dbt Semantic Layer with Python. It allows developers to interact with the Semantic Layer APIs and query metrics and dimensions in downstream tools.
+
+## Installation
+
+To install the SDK, you'll need to specify optional dependencies depending on whether you want to use it synchronously (backed by [requests](https://github.com/psf/requests/)) or with asynchronous ([asyncio](https://docs.python.org/3/library/asyncio.html) backed by [aiohttp](https://github.com/aio-libs/aiohttp/)).
+
+<Tabs>
+<TabItem value="sync" label="Sync installation">
+
+Sync installation means your program waits for each task to finish before moving on to the next one. 
+
+It's simpler and easier to understand. Suitable for smaller tasks or when your program doesn't need to handle many tasks at the same time.
+
+```bash
+pip install dbt-sl-sdk[sync]
+```
+</TabItem>
+
+<TabItem value="async" label="Async installation">
+
+Async installation means your program can start a task and then move on to other tasks while waiting for the first one to finish.
+
+It can handle many tasks at once without waiting, making it faster and more efficient for larger tasks or when you need to manage multiple tasks at the same time.
+
+```bash
+pip install dbt-sl-sdk[async]
+```
+</TabItem>
+</Tabs>
+
+## Usage
+To run operations against the Semantic Layer APIs, instantiate (create an instance of) a `SemanticLayerClient` with your specific [API connection parameters](/docs/dbt-cloud-apis/sl-api-overview):
+
+```python
+from dbtsl import SemanticLayerClient
+
+client = SemanticLayerClient(
+    environment_id=123,
+    auth_token="<your-semantic-layer-api-token>",
+    host="semantic-layer.cloud.getdbt.com",
+)
+
+# query the first metric by `metric_time`
+def main():
+    with client.session():
+        metrics = client.metrics()
+        table = client.query(
+            metrics=[metrics[0].name],
+            group_by=["metric_time"],
+        )
+        print(table)
+
+main()
+```
+
+**Note**: All method calls that reach out to the APIs need to be within a `client.session()` context manager. This allows the client to establish a connection to the APIs only once and reuse the same connection between API calls.
```

Comment:
I'd add that we discourage creating a session per request since that's very inefficient. Ideally, they should create an application-wide session, and reuse the same session throughout the app for optimal performance.